opentelemetry-sdk/src/opentelemetry/sdk/_metrics/data.py

```diff
@@ -0,0 +1,35 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from opentelemetry.sdk.resources import Resource
+from opentelemetry.sdk.util.instrumentation import InstrumentationInfo
+
+
+class Metric:
```

Comment:
Not a huge deal but both `LogRecord` and `ReadableSpan` are within the `__init__.py` files instead of separated. Should we be consistent?

Reply:
I ended up in a circular dependency because of https://github.com/open-telemetry/opentelemetry-python/blob/8543481a40e006d55ff46316609556023a6ce7b4/opentelemetry-sdk/src/opentelemetry/sdk/_metrics/__init__.py#L103

__init__.py imports the exporter and the exporter imports init.py if the data classes are in it.

Looks like both log/trace init aren't importing the exporters 🤔

Reply:
I propose fixing this in a follow up PR, since the MetricData is just a stub which will be addressed in https://github.com/open-telemetry/opentelemetry-python/issues/2327

Reply:
@codeboten 
I think this is another reason why we should remove exporters from the `Provider` interface. See [this](https://github.com/open-telemetry/opentelemetry-python/issues/2316) issue. Providers should communicate with exporters and propagate config through the `MetricReader`s, not directly.

In any case, maybe just remove the typing annotation from `__init__.py`? Or will mypy complain? If so, I'm okay with leaving the code as is and we can address it after the above issue is addressed.

Reply:
I added a note to the issue you linked, i think that's probably the right move. I can remove it since mypy doesnt run on the SDK, but i'd rather leave the annotation in and move the code later.